s3store/s3store.go

```diff
@@ -517,6 +501,33 @@ func (store S3Store) ConcatUploads(dest string, partialUploads []string) error {
 	return store.FinishUpload(dest)
 }
 
+func (store S3Store)ListAllParts(id string) (parts []*s3.Part, err error) {
```

Comment:
Please rename this function to `listAllParts`. The lower-cased letter will cause this function to not be available to other packages which is what we want from a private function.